File: tutorial_1/sample_2/main.cpp
#include <filesystem>
#include <iostream>

#include "main.hpp"

namespace fs = std::experimental::filesystem;

int main(int argc, char* argv[]) {

    cv::Mat image = cv::Mat(cv::Size(640, 480), CV_8UC3, cv::Scalar(255, 0, 0));

    cv::imshow("image", image);

    cv::waitKey(0);

    fs::path executablePath{ fs::path(argv[0]) };

	fs::path dataPath{ fs::path("/data") };

	fs::path outputPath{ executablePath.parent_path().append(dataPath) };

    if(!fs::is_directory(outputPath)) {

        fs::create_directory(outputPath);

    }

    fs::path outputFile {outputPath.append(fs::path("/image.png"))};

	std::cout << outputFile.string() << std::endl;

    cv::imwrite(outputFile.string(), image);
    
    return 0;
}

File: tutorial_1/sample_2/main.py
import cv2 as cv
import numpy as np

from absl import app
from pathlib import Path

def main(argv):

    image = (255, 0, 0)*np.ones((480, 640, 3))

    cv.imshow("image", image)

    cv.waitKey(0)

    executable_path = Path(argv[0])

    output_path = (executable_path.parent).joinpath("data")

    if (not Path.exists(output_path)):

        Path.mkdir(output_path)

    output_file = output_path.joinpath("image.png")

    print(str(output_file))

    cv.imwrite(str(output_file), image)

    cv.destroyAllWindows()

    return 0


if __name__ == "__main__":

    app.run(main)